include/zephyr/bluetooth/gatt.h

```diff
@@ -377,6 +408,17 @@ struct bt_gatt_cpf {
  */
 void bt_gatt_cb_register(struct bt_gatt_cb *cb);
 
+/** @brief Register GATT authorization callbacks.
+ *
+ *  Register callbacks to perform application-specific authorization of
+ *  GATT operations.
+ *
+ *  @param cb Callback struct.
+ *
+ *  @return Zero on success or negative error code otherwise
+ */
+int bt_gatt_authorization_cb_register(const struct bt_gatt_authorization_cb *cb);
```

Comment:
can you document that you first have to clear the cb by passing in NULL? E.g. if you want to change the current cb to point to something else.

Also this is missing something about why this is necessary.
I would change L413-L414 to
```
 *  Register callbacks to perform application-specific authorization of
 *  GATT operations on all characteristics and UUIDs.
```

Reply:
Sure, I will improve the API documentation with the next patch. I also forgot to mention here that you need to enable the `CONFIG_BT_GATT_AUTHORIZATION_CUSTOM` Kconfig to use this API